cylc/flow/config.py

```diff
@@ -2091,19 +2105,19 @@ def get_graph_raw(
                         l_point = point
                     l_id = (name, l_point)
 
-                    if l_id is None and r_id is None:
-                        continue
-                    if l_id is not None and actual_first_point > l_id[1]:
+                    if actual_first_point > l_point:
                         # Check that l_id is not earlier than start time.
-                        if (r_id is None or r_id[1] < actual_first_point or
```

Comment:
Re-arranged the conditions in this `if` statement so that the expensive `r_id[1] < actual_first_point` check comes last (so will only be performed if absolutely necessary).